connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java

```diff
@@ -325,6 +325,29 @@ public SchemaBuilder field(String fieldName, Schema fieldSchema) {
     }
 
     /**
+     * Add a field to the struct schema by name, that will be resolved later. Throws a SchemaBuilderException if this
+     * is not a struct schema.
+     * @param fieldName the name of the field to add
+     * @param schemaName the name of the schema for the field
+     * @param optional a flag if the schema is optional or not
+     * @return the SchemaBuilder
+     */
+    public SchemaBuilder field(String fieldName, String schemaName, boolean optional) {
```

Comment:
Wouldn't you also need to handle default values? Presumably you're also assuming that there's only one version of a schema referenced anywhere in the schema? And that the schema parameters (e.g. the scale used by Decimal) don't matter? I realize that Decimal isn't a good example of where this could come into play, I'm just trying to highlight that I'm not sure the current API really addresses the full problem.